dockers/docker-dhcp-relay/dhcpv6-relay.agents.j2

```diff
@@ -1,28 +1,31 @@
-{# Append DHCPv6 agents #}
-{% if VLAN and vlan_name in VLAN and 'dhcpv6_servers' in VLAN[vlan_name] and VLAN[vlan_name]['dhcpv6_servers']|length > 0 %}
-{% for dhcpv6_server in VLAN[vlan_name]['dhcpv6_servers'] %}
-{% if dhcpv6_server | ipv6 %}
-{% set _dummy = relay_for_ipv6.update({'flag': True}) %}
-{% endif %}
-{% endfor %}
-{% if relay_for_ipv6.flag %}
-{% set _dummy = relay_for_ipv6.update({'flag': False}) %}
-[program:isc-dhcpv6-relay-{{ vlan_name }}]
-{# We treat this VLAN as a downstream interface (-l), as we only want to listen for requests #}
-command=/usr/sbin/dhcrelay -d -6 --name-alias-map-file /tmp/port-name-alias-map.txt -l {{ vlan_name }}
-{#- We treat all other interfaces as upstream interfaces (-u), as we only want to listen for replies #}
+{# Append DHCPv6 agent #}
+{% set relay_for_ipv6 = { 'flag': False } %}
+[program:isc-dhcpv6-relay-{{ v6_vlan_list|join("-") }}]
+command=/usr/sbin/dhcrelay -d -6 --name-alias-map-file /tmp/port-name-alias-map.txt
+{#- We get all DHCPv6 servers only once #}
+{%- set dhcpv6_server_list = [] %}
+{%- for vlan_name in v6_vlan_list %}
 {%- for dhcpv6_server in VLAN[vlan_name]['dhcpv6_servers'] %}
+{%- if dhcpv6_server | ipv6 and dhcpv6_server not in dhcpv6_server_list %}
+{%- set _dummy = dhcpv6_server_list.append( dhcpv6_server ) %}
+{%- endif %}
+{%- endfor %}
+{%- endfor %}
+{#- We treat each VLAN as a downstream interface (-l), as we only want to listen for requests #}
+{%- for vlan_name in v6_vlan_list %} -l {{ vlan_name }}{% endfor -%}
+{#- We treat all other interfaces as upstream interfaces (-u), as we only want to listen for replies #}
+{%- for dhcpv6_server in dhcpv6_server_list %}
 {%- if dhcpv6_server | ipv6 %}
 {%- for (name, prefix) in VLAN_INTERFACE|pfx_filter %}
-{%- if prefix | ipv6 and name != vlan_name %} -u {{ dhcpv6_server }}%%{{ name }} {% endif -%}
+{%- if prefix | ipv6  %} -u {{ dhcpv6_server }}%%{{ name }} {% endif -%}
```

Comment:
@shlomibitton is this by design to treat downstream interfaces as upstream as well?

Reply:
if it is a L3 interface so yes

Reply:
@shlomibitton suggest to double check whether its corresponding to the design

Reply:
Verified and approved.